xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-default/src/test/java/org/xwiki/user/internal/UserReferenceConverterTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.user.internal;
+
+import java.lang.reflect.Type;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+import org.xwiki.user.UserReference;
+import org.xwiki.user.UserReferenceResolver;
+
+import com.xpn.xwiki.XWikiContext;
+
+/**
+ * Test of {@link UserReferenceConverter}.
+ *
+ * @version $Id$
+ * @since 12.8RC1
+ */
+@ComponentTest
+class UserReferenceConverterTest
+{
+    @InjectMockComponents
+    private UserReferenceConverter converter;
+
+    @MockComponent
+    private UserReferenceResolver<String> userReferenceResolver;
+
+    @MockComponent
+    private Provider<XWikiContext> xcontextProvider;
+
+    @Test
+    void convertToType()
+    {
+        XWikiContext context = mock(XWikiContext.class);
+        when(this.xcontextProvider.get()).thenReturn(context);
+        when(context.getWikiReference()).thenReturn(new WikiReference("s1"));
+
+        this.converter.convertToType(mock(Type.class), "XWiki.U1");
+
+        verify(this.userReferenceResolver).resolve("XWiki.U1", new WikiReference("s1"));
```

Comment:
I dont understand why you use `verify()` instead of doing an `assert` on the result.